src/script/sign.h

```diff
@@ -68,6 +71,434 @@ struct SignatureData {
+                        }
+
+                        // Read in the signature from value
+                        std::vector<unsigned char> sig;
+                        sig.resize(value_len);
+                        Span<unsigned char> sig_data = MakeSpan(sig);
+                        s >> sig_data;
+
+                        // Add to list
+                        input.partial_sigs.emplace(pubkey, sig);
+                    }
+                    break;
+                // BIP 32 HD Keypaths and sighash types
+                case PSBT_BIP32_KEYPATH_SIGHASH:
+                    if (in_globals) {
+                        // Make sure that the key is the size of pubkey + 1
+                         if (key.size() != CPubKey::PUBLIC_KEY_SIZE + 1 && key.size() != CPubKey::COMPRESSED_PUBLIC_KEY_SIZE + 1) {
+                            throw std::ios_base::failure("Size of key was not the expected size for the type BIP32 keypath");
+                        }
+                        // Read in the pubkey from key
+                        CPubKey pubkey(key.begin() + 1, key.end());
+                        if (!pubkey.IsFullyValid()) {
+                           throw std::ios_base::failure("Invalid pubkey");
+                        }
+
+                        // Read in key path
+                        std::vector<uint32_t> keypath;
+                        for (unsigned int i = 0; i < value_len; i += sizeof(uint32_t)) {
+                            uint32_t index;
+                            s >> index;
+                            keypath.push_back(index);
+                        }
+
+                        // Add to map
+                        hd_keypaths.emplace(pubkey, keypath);
+                    } else {
+                        // Read in the sighash type
+                        s >> input.sighash_type;
+                    }
+                    break;
+                // Number of inputs and input index
+                case PSBT_NUM_IN_VIN:
+                    if (in_globals) {
+                        num_ins = ReadCompactSize(s);
+                    } else {
+                        // Make sure that we are using inout indexes or this is the first input
+                        if (!use_in_index && separators != 1) {
+                            throw std::ios_base::failure("Input indexes being used but an input does not provide its index");
+                        }
+
+                        input.index = ReadCompactSize(s);
+                        use_in_index = true;
+                        input.use_in_index = true;
+                    }
+                    break;
+                // Unknown stuff
+                default:
+                    // Read in the value
+                    std::vector<unsigned char> val_bytes;
+                    val_bytes.resize(value_len);
+                    Span<unsigned char> value = MakeSpan(val_bytes);
+                    s >> value;
```

Comment:
In "Implement PSBT structures and un/serliazation methods per BIP 174":

I think this (and other places where `Span` is used) would look less confusing if you did `s >> MakeSpan(val_bytes)`, since you are not using `value`, and it looks like the data is written to some unrelated thing (i.e. not `val_bytes`, but `value`). Not sure if that is possible though (it should be, if it isn't, but that's a separate story).

Reply:
Done here and everywhere else this happens.